### PR TITLE
ITE: drivers/i2c: Modify I2C related settings

### PR DESCRIPTION
**Pull request by GTLin08**
This pull request includes two changes:
1. Add a property to adjust the I2C data hold time which will pass the SI test.
2. Adjust the prescale of I2C SCL low and high period.

### CODE REVIEW

File: drivers/i2c/i2c_ite_enhance.c [hunk 1]
```diff
@@ -296,6 +296,7 @@ static void i2c_enhanced_port_set_frequency(const struct device *dev,
 	const struct i2c_enhance_config *config = dev->config;
 	uint32_t clk_div, psr, pll_clock;
 	uint8_t *base = config->base;
+	uint8_t psr_h, psr_l;
```
Comment (keith-zephyr) [inline drivers/i2c/i2c_ite_enhance.c:299]: These should be 32-bit values to avoid overflow issues.

File: drivers/i2c/i2c_ite_enhance.c [hunk 2]
```diff
@@ -318,11 +319,21 @@ static void i2c_enhanced_port_set_frequency(const struct device *dev,
...
 
 		/* Adjust SCL low period prescale */
-		psr += config->prescale_scl_low;
+		psr_l = psr + config->prescale_scl_low;
+		if (psr_l > 0xFD) {
+			psr_l = 0xFD;
+		}
+
+		/* Adjust SCL high period prescale */
+		psr_h = psr - config->prescale_scl_low;
```
Comment (keith-zephyr) [inline drivers/i2c/i2c_ite_enhance.c:328]: This calculation could underflow.  You should verify that `config->prescale_scl_low` is less than `psr`.

File: dts/bindings/i2c/ite,enhance-i2c.yaml [hunk 3]
```diff
@@ -18,6 +18,30 @@ properties:
...
+      Time calculation is as follows.
+      Note: T = 1/smb_clk (sec)
+      0 = 3T smb_clk,
+      1 = (3+1)T smb_clk,
+      2 = (3+2)T smb_clk,
+      3 = (3+3)T smb_clk,
+      4 = (3+4)T smb_clk,
+      5 = (3+5)T smb_clk,
+      6 = (3+6)T smb_clk,
+      7 = (3+7)T smb_clk,
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply dts/bindings/i2c/ite,enhance-i2c.yaml]: Done

File: drivers/i2c/i2c_ite_enhance.c (hunk 1 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_enhance.c]: Done.

File: drivers/i2c/i2c_ite_enhance.c (hunk 2 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_enhance.c]: Done.

File: drivers/i2c/i2c_ite_enhance.c [hunk 4]
```diff
@@ -1196,6 +1198,14 @@ static int i2c_enhance_init(const struct device *dev)
 				(IT8XXX2_SMB_SMB45CHS &= ~GENMASK(6, 4));
 		}
 
+		/* The property setting range of data hold time is 3-10. */
+		if (data_hold_time < 3) {
+			data_hold_time = 3;
```
Comment (keith-zephyr) [inline drivers/i2c/i2c_ite_enhance.c:1220]: This check is not needed. The devicetree compiler will generate an error if the user specifies a `data-hold-time` outside the enum list.

File: drivers/i2c/i2c_ite_enhance.c [hunk 5]
```diff
@@ -318,11 +319,27 @@ static void i2c_enhanced_port_set_frequency(const struct device *dev,
 		}
 
 		/* Adjust SCL low period prescale */
-		psr += config->prescale_scl_low;
+		psr_l = psr + prescale_scl;
+		if (psr_l > 0xFD) {
+			psr_l = 0xFD;
```
Comment (keith-zephyr) [inline drivers/i2c/i2c_ite_enhance.c:324]: This path should probably also have a warning.

Comment (GTLin08) [reply drivers/i2c/i2c_ite_enhance.c:324]: Done.

File: drivers/i2c/i2c_ite_enhance.c (hunk 4 above)
Comment (GTLin08) [reply drivers/i2c/i2c_ite_enhance.c]: Removed and added defult value in yaml.

File: dts/bindings/i2c/ite,enhance-i2c.yaml [hunk 6]
```diff
@@ -18,6 +18,30 @@ properties:
...
+      Time calculation is as follows.
+      Note: T = 1/smb_clk (sec)
+      3 = 3T smb_clk,
+      4 = (3+1)T smb_clk,
+      5 = (3+2)T smb_clk,
+      6 = (3+3)T smb_clk,
+      7 = (3+4)T smb_clk,
+      8 = (3+5)T smb_clk,
+      9 = (3+6)T smb_clk,
+      10 = (3+7)T smb_clk,
```
[the remark this answers is not in the document]
Comment (GTLin08) [reply dts/bindings/i2c/ite,enhance-i2c.yaml]: Done.

**Review by keith-zephyr — APPROVED** [review]

**Review by Dino-Li — APPROVED** [review]

**Pull request merged**